src/runner/kafka.js

```diff
@@ -0,0 +1,58 @@
+// @flow
+import Kafka from 'no-kafka';
+import type { IRunner, Configuration, Logger, Consumer, IRegistry } from '../../types';
+
+class KafkaRunner implements IRunner {
+  config: Configuration;
+  logger: Logger;
+  registry: IRegistry;
+  consumer: Consumer;
+
+  constructor(config: Configuration, registry: IRegistry, logger: Logger) {
+    this.config = config;
+    this.registry = registry;
+    this.logger = logger;
+
+    this.consumer = new Kafka.GroupConsumer({
+      groupId: config.kafkaGroupId,
+      clientId: config.clientId,
+      connectionString: config.kafkaConnection,
+      codec: config.kafkaCodec,
+      logger: {
+        logLevel: config.logLevel,
+      },
+    });
+  }
+
+  receive = async (messages: Array<Object>, topic: string, partition: string) => {
+    for (const m of messages) { // eslint-disable-line no-restricted-syntax
+      let params: Object = {};
+      try {
+        // commit offset
+        params = JSON.parse(m.message.value.toString('utf8'));
+
+        this.registry.events.emit('runner_receive', topic, params);
+        await this.consumer.commitOffset({ topic, partition, offset: m.offset, metadata: 'optional' }); // eslint-disable-line
```

Comment:
I'd really encourage getting this done sooner rather than later. It's a really straightforward thing to implement. That article is useful, but does overcomplicate what's a pretty simple concept (a Java programmer overcomplicating things? I have never heard of such a thing.)

This could be literally a 2 line fix:

```
const commitOffset = () => this.consumer.commitOffset({ topic, partition, offset: m.offset, metadata: 'optional' });

const task = this.registry.getTask(topic); 

if (task.config.mode == 'at_most_once') await commitOffset();

await task.subscribe(params);

if (task.config.mode == 'at_least_once') await commitOffset();
```

Job done.